src/test/scala/mesosphere/marathon/integration/UpgradeIntegrationTest.scala

```diff
@@ -0,0 +1,236 @@
+
+    override val processBuilder = {
+      val bin = new File(marathonPackage, "marathon-1.5.6/bin/marathon").getCanonicalPath
+      val cmd = Seq("bash", bin, "-J-Xmx1024m", "-J-Xms256m", "-J-XX:+UseConcMarkSweepGC", "-J-XX:ConcGCThreads=2") ++ akkaJvmArgs ++
+        Seq(s"-DmarathonUUID=$uuid -DtestSuite=$suiteName") ++ args
+      Process(cmd, workDir, sys.env.toSeq: _*)
+    }
+  }
+
+  case class Marathon16322(suiteName: String, masterUrl: String, zkUrl: String)(
+      implicit
+      val system: ActorSystem, val mat: Materializer, val ctx: ExecutionContext, val scheduler: Scheduler) extends BaseMarathon with MarathonDownload {
+
+    override val marathonPackage = Files.createTempDirectory("marathon-1.6.322").toFile
+    override val tarballName = "marathon-1.6.322.tgz"
+    override val downloadURL = new URL("https://downloads.mesosphere.com/marathon/releases/1.6.322/marathon-1.6.322-2bf46b341.tgz")
+
+    override val processBuilder = {
+      val bin = new File(marathonPackage, "marathon-1.6.322-2bf46b341/bin/marathon").getCanonicalPath
+      val cmd = Seq("bash", bin, "-J-Xmx1024m", "-J-Xms256m", "-J-XX:+UseConcMarkSweepGC", "-J-XX:ConcGCThreads=2") ++ akkaJvmArgs ++
+        Seq(s"-DmarathonUUID=$uuid -DtestSuite=$suiteName") ++ args
+      Process(cmd, workDir, sys.env.toSeq: _*)
+    }
+  }
+
+  "Ephemeral and persistent apps and pods" should {
+    "survive an upgrade cycle" taggedAs WhenEnvSet(envVarRunMesosTests, default = "true") in {
+
+      // Start apps in 1.4.9
+      Given("A Marathon 1.4.9 is running")
+      marathon149.start().futureValue
+      (marathon149.client.info.entityJson \ "version").as[String] should be("1.4.9")
+
+      And("new running apps in Marathon 1.4.9")
+      val app_149_fail = appProxy(testBasePath / "app-149-fail", "v1", instances = 1, healthCheck = None)
+      marathon149.client.createAppV2(app_149_fail) should be(Created)
+
+      val app_149 = appProxy(testBasePath / "app-149", "v1", instances = 1, healthCheck = None)
+      marathon149.client.createAppV2(app_149) should be(Created)
+
+      eventually { marathon149 should have (runningTasksFor(app_149.id.toPath, 1)) }
+      eventually { marathon149 should have (runningTasksFor(app_149_fail.id.toPath, 1)) }
+
+      val originalApp149Tasks = marathon149.client.tasks(app_149.id.toPath).value
+      val originalApp149FailedTasks = marathon149.client.tasks(app_149_fail.id.toPath).value
+
+      When("Marathon 1.4.9 is shut down")
+      marathon149.stop().futureValue
+
+      And(s"App ${app_149_fail.id} fails")
+      killTask("app-149-fail")
+
+      // Pass upgrade to 1.5.6
+      And("Marathon is upgraded to 1.5.6")
+      marathon156.start().futureValue
+      (marathon156.client.info.entityJson \ "version").as[String] should be("1.5.6")
+
+      And("new apps in Marathon 1.5.6 are added")
+      val app_156 = appProxy(testBasePath / "app-156", "v1", instances = 1, healthCheck = None)
```

Comment:
Please give them apps longer and more expressive names e.g. `migration-app-156-with-persistent-volumes`. Makes it much easier later to debug and reduces the chance of app name collision.